client/client_test.go

```diff
@@ -95,17 +95,17 @@ func generateRedisProbeDef(name string) *probes_configpb.ProbeDef {
 
 // TestRegisterAndAddProbe tests that an extension probe type can be registered and added to Cloudprober without error.
 func TestRegisterAndAddProbe(t *testing.T) {
-	ctx, client := setupCloudproberAndClient(t)
-	defer teardownCloudproberAndClient(ctx, t, client)
+	client := setupCloudproberAndClient(t)
+	defer teardownCloudproberAndClient(t, client)
 
 	for i := 0; i < testRunCount; i++ {
 		probeName := fmt.Sprintf("testExtension%d", i)
-		if err := client.RegisterAndAddProbe(ctx, int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName), &myprobe.Probe{}); err != nil {
+		if err := client.RegisterAndAddProbe(context.Background(), int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName), &myprobe.Probe{}); err != nil {
```

Comment:
Why are we using a package from the "examples" dir?

Reply:
At the time of writing this test, we hadn't started writing an extension probe. This example provides a working extension probe implementation. There is a TODO to change this to our probe implementation when it is ready.